test/runner.js

```diff
@@ -0,0 +1,192 @@
+'use strict'
+
+let fs = require('fs')
+let path = require('path')
+let expect = require('chai').expect
+let rimraf = require('rimraf')
+let Launcher = require('webdriverio/build/lib/launcher')
+let parseXmlString = require('xml2js').parseString
+
+let configFile = './test/fixtures/wdio.conf.js'
+let resultsDir = path.join(__dirname, '../allure-results') 
+
+describe('"before all" hook', () => {
+
+  beforeEach(() => {
+    clean()
+  })
+
+  it('should not show up in the results when it is passing', () => {
+
+    return run(['before-all-passing']).then((code) => {
+
+      return getResultsXML()
```

Comment:
`then` inside another `then` callback is usually an anti-pattern

I'd recomend to write this:
```js
return run(['before-all-passing']).then((code) => {
   expect(code).to.equal(1) // you don't check actual exit code, but I think it should be done
   return getResultsXML()
}).then((results) => {
  // your checks here
})
```